src/components/NavBar.tsx

```diff
@@ -7,22 +6,36 @@ import {
   faTools,
 } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import tools from '../assets/tools.json';
 
 import './NavBar.css';
 import OfekLogo from '../assets/images/OfekLogo.png';
 
-interface PropsType {
-  toggleMainWidth(): void;
+interface NavBarPropsType {
+  toggleMainWidth:() => void;
   mini: Boolean;
 }
-const NavBar = (props: PropsType) => {
+const NavBar = ({ toggleMainWidth, mini }: NavBarPropsType) => {
   const SideBarElement: HTMLElement | null = document.getElementById('sidebar');
```

Comment:
Please use `useRef` instead of referencing directly to the DOM.

Here's a really good tutorial for this important hook: [useRef tutorial](https://www.youtube.com/watch?v=t2ypzz6gJm0)